src/site/_includes/components/Meta.js

```diff
@@ -0,0 +1,116 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const path = require("path");
+const site = require("../../_data/site");
+const stripLanguage = require("../../_filters/strip-language");
+const {html} = require("common-tags");
+
+module.exports = (locale, page, collections) => {
+  const pageData = collections.all.find((item) => item.url === page.url).data;
+  const pageUrl = stripLanguage(page.url);
+
+  /**
+   * Find post meta data associated with a social media platform.
+   * Post authors should be able to create a `social` object in their YAML
+   * frontmatter and specify a unique title, description, and image for each
+   * social media platform.
+   * Supported platforms are Google (mostly for Search cards),
+   * Facebook Open Graph, and Twitter.
+   * @param {!string} platform The name of a social media platform.
+   * @return {Object} Meta data object for the social media platform specified.
+   */
+  function getMetaByPlatform(platform) {
+    const social =
+      pageData.social && pageData.social[platform]
+        ? pageData.social[platform]
+        : {};
+
+    const title = social.title || pageData.title;
+    const description = social.description || pageData.description;
+    let thumbnail = social.thumbnail || pageData.thumbnail || pageData.hero;
+    const alt = social.alt || pageData.alt || site.name;
+
+    // If the page doesn't have social media images, a hero, or a thumbnail,
+    // fallback to using the site's default thumbnail.
+    // Return a full path to the image using our image CDN.
+    if (!thumbnail) {
+      thumbnail = new URL(site.thumbnail, site.imageCdn);
+    } else {
+      thumbnail = new URL(path.join(pageUrl, thumbnail), site.imageCdn);
+    }
+    thumbnail.searchParams.set("auto", "format");
+    thumbnail.searchParams.set("fit", "max");
+    thumbnail.searchParams.set("w", 1200);
+    thumbnail = thumbnail.toString();
+
+    return {title, description, thumbnail, alt};
+  }
+
+  function renderGoogleMeta() {
+    const meta = getMetaByPlatform("google");
+    return html`
+      <meta itemprop="name" content="${meta.title}" />
+      <meta itemprop="description" content="${meta.description}" />
+      <meta itemprop="image" content="${meta.thumbnail}" />
+    `;
+  }
+
+  function renderFacebookMeta() {
+    const meta = getMetaByPlatform("facebook");
+    // nb. This will mark pages like /404/ and /offline/ as "article" but that's
+    // probably fine as those shouldn't show up in search results anyway.
+    const type = pageUrl === "/" ? "website" : "article";
+    // Filter out tags that we don't want to show up in Search.
+    // These tags are only used internally to determine which layout a page
+    // should use.
+    let tags = pageData.tags || [];
```

Comment:
I think this might be clearer if you start off with

```js
let tags = (type === "article" ? pageData.tags : null) || [];
```

(or similar). The block below can just be the mapping and I can see more clearly that tags are not allowed for the top-level.